docker/Dockerfile.concrete-compiler-env

```diff
@@ -24,9 +28,9 @@ RUN ./bootstrap.sh && ./b2 --with-filesystem install
 COPY --from=ghcr.io/zama-ai/hpx:latest /hpx /hpx
 ENV HPX_INSTALL_DIR=/hpx/build
 # Setup CUDA
-COPY --from=ghcr.io/zama-ai/cuda:12-3 /usr/local/cuda-12.3/ /usr/local/cuda-12.3/
-COPY --from=ghcr.io/zama-ai/cuda:12-3 /usr/lib64/libcuda.so* /usr/lib64/
-ENV PATH "$PATH:/usr/local/cuda-12.3/bin"
+COPY --from=ghcr.io/zama-ai/cuda:11-8 /usr/local/cuda-11.8/ /usr/local/cuda-11.8/
+COPY --from=ghcr.io/zama-ai/cuda:11-8 /usr/lib64/libcuda.so* /usr/lib64/
+ENV PATH "$PATH:/usr/local/cuda-11.8/bin"
```

Comment:
Not sure why this is done - does this mean that the compiler is compiled against cuda11.8 only?

Reply:
Maybe we should think about it differently in the future, but for now, the compiler env is using a specific cuda version, so that one should be cuda118 for the release. If we want to later have multiple cuda versions in the CI, then we would need to change how we build and use the compiler image